tests/field_deconstruction/tests.py

```diff
@@ -219,6 +220,8 @@ def test_foreign_key(self):
 
     @override_settings(AUTH_USER_MODEL="auth.Permission")
     def test_foreign_key_swapped(self):
+        # Clear cache for mapping models to swappable setting
+        apps.get_swappable_setting.cache_clear()
```

Comment:
Seems like a context manager would be good here so the cache can be cleared regardless of whether or not the test passes (also we could simply decorate the test method).